lib/kubernetes-deploy/runner.rb

```diff
@@ -76,11 +78,14 @@ def run
       phase_heading("Parsing deploy content")
       resources = discover_resources
 
+      tprs = resources.reject! { |r| r.tpr? }
+
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
       phase_heading("Predeploying priority resources")
       predeploy_priority_resources(resources)
+      update_tprs(tprs)
```

Comment:
We don't necessarily want `update` to be used for non-TPRs (it can be destructive as it's a wholesale replacement of the resource, it doesn't do any merge strategy magic).

Right now, the `deploy_resources` appears to create a single `apply` for all resources. Is this intended? This fix would be easy if the `apply` line was distinct per resource, but the implementation is currently:

``` ruby
    def deploy_resources(resources, prune: false)
      command = ["apply", "--namespace=#{@namespace}"]
      KubernetesDeploy.logger.info("Deploying resources:")

      resources.each do |r|
        KubernetesDeploy.logger.info("- #{r.id}")
        command.push("-f", r.file.path)
        r.deploy_started = Time.now.utc
      end

      if prune
        command.push("--prune", "--all")
        PRUNE_WHITELIST.each { |type| command.push("--prune-whitelist=#{type}") }
      end

      run_kubectl(*command)
    end
```